src/main/java/com/maxmind/minfraud/WebServiceClient.java

```diff
@@ -319,99 +354,96 @@ public void reportTransaction(TransactionReport transaction) throws IOException,
         if (transaction == null) {
             throw new IllegalArgumentException("transaction must not be null");
         }
-        URL url = createUrl(WebServiceClient.pathBase + service);
-        HttpPost request = requestFor(transaction, url);
+        URI uri = createUri(WebServiceClient.pathBase + service);
+        HttpRequest request = requestFor(transaction, uri);
+
+        HttpResponse<InputStream> response = null;
+        try {
+            response = httpClient.send(request, HttpResponse.BodyHandlers.ofInputStream());
+        } catch (InterruptedException e) {
+            throw new MinFraudException("Interrupted sending request", e);
 
-        try (CloseableHttpResponse response = httpClient.execute(request)) {
-            return handleResponse(response, url, cls);
         }
+        return handleResponse(response, uri, cls);
```

Comment:
Should we call `exhaustBody()` somehow here? And/or close the body?

Reply:
Woops, good catch. I don't think there are any additional cases where `exhaustBody()` needs to be called for that code path, but we should be calling close.